pageserver/src/metrics.rs

```diff
@@ -497,6 +497,24 @@ pub(crate) static WAIT_LSN_IN_PROGRESS_GLOBAL_MICROS: Lazy<IntCounter> = Lazy::n
     .expect("failed to define a metric")
 });
 
+pub(crate) static ONDEMAND_DOWNLOAD_BYTES: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "pageserver_ondemand_download_bytes_total",
+        "Total bytes of layers on-demand downloaded",
+        &["task_kind"]
+    )
+    .expect("failed to define a metric")
+});
+
+pub(crate) static ONDEMAND_DOWNLOAD_COUNT: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "pageserver_ondemand_download_count",
```

Comment:
nit: Prometheus metric naming convension advises using `_total` suffix "for a unit-less accumulating count". So `pageserver_ondemand_downloads_total` would be a more canonicial name.
But our codebase already uses `_count` suffix along with `_total` here and there, so it's fine either way.

https://prometheus.io/docs/practices/naming/